core/views.py

```diff
@@ -1,22 +1,88 @@
-from rest_framework import permissions
-from rest_framework import generics
-from rest_framework import status
-from .serializers import HackathonSerializer
-from .models import Hackathon
-from .permissions import HackathonPermissions
+from rest_framework import generics, status, permissions, exceptions
+from rest_framework.response import Response
+from django.utils import timezone
+from django.utils.decorators import method_decorator
+from drf_yasg.utils import swagger_auto_schema
+from drf_yasg import openapi
+from .models import Hackathon, Team
+from .serializers import HackathonTeamCreateSerializer, HackathonSerializer, TeamSerializer
+from .permissions import HackathonPermissions, isProfileComplete
+
+class HackathonTeamView(generics.GenericAPIView):
+    # Required for POST request
+    serializer_class = HackathonTeamCreateSerializer
+    permission_classes = [isProfileComplete]
+
+    # To over-write get_queryset, removes AssertionError
+    def get_queryset(self):
+        return
+
+    def post(self, request, *args, **kwargs):
+        """
+        Users with completed profile can create a team with POST request
+        """
+        if request.data['hackathon'] != kwargs['pk']:
+            raise exceptions.ValidationError("Hackathon ID does not match in URL and Request body!")
+        serializer = self.get_serializer(data=request.data)
+        serializer.is_valid(raise_exception=True)
+        serializer.save()
+        return Response(serializer.data, status.HTTP_200_OK)
+
+    def get(self, request, *args, **kwargs):
+        """
+        Returns list of team registered for a hackathon
+        """
+        try:
+            if Hackathon.objects.get(id=kwargs['pk']) is not None:
+                queryset = Team.objects.filter(hackathon=kwargs['pk'])
+        except:
+            return Response("Hackathon does not exist!", status=status.HTTP_404_NOT_FOUND)
+        serializer = TeamSerializer(queryset, many=True)
+        return Response(serializer.data)
```

Comment:
You can write it in a better way. Maybe refer https://github.com/codefest-iit-bhu/codefest-app/blob/0e7c8f737ca684797ed512931870094c9b547ab7/website/views.py#L92